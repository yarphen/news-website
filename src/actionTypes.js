const keyMirror = obj => Object.keys(obj).reduce((prev, key) => ({ ...prev, [key]: key }), {});

export default keyMirror({
  PING_REQUEST_START: 0,
  PING_REQUEST_SUCCESS: 0,
  PING_REQUEST_FAIL: 0,
  PING_RESET: 0,
});
