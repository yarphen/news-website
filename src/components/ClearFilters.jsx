/* eslint-disable jsx-a11y/click-events-have-key-events */
import React from 'react';
import PropTypes from 'prop-types';

export const ClearFilters = ({ onClearFilters }) => (
  <div className="clear-filters">
    <div className="clear-filters-btn" onClick={onClearFilters}>Clear Filters</div>
  </div>
);


ClearFilters.propTypes = {
  onClearFilters: PropTypes.func.isRequired,
};
