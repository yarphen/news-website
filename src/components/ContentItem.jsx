/* eslint-disable jsx-a11y/anchor-is-valid */
import React from 'react';
import PropTypes from 'prop-types';

import { formatDate } from '../util/format';

export const ContentItem = (props) => {
  const {
    title, text, image, updated,
  } = props;
  return (
    <div className="content-item">
      <div className="content-item-desktop-title"><a href="#">{title}</a></div>
      <div className="content-item-image"><a href="#"><img src={image} /></a></div>
      <div className="content-item-article">
        <div className="content-item-mobile-title"><a href="#">{title}</a></div>
        <div className="content-item-text">{text}</div>
        <div className="content-item-updated">
          <span>Updated: </span>
          <span>{formatDate(updated)}</span>
        </div>
      </div>
    </div>
  );
};

ContentItem.propTypes = {
  title: PropTypes.string.isRequired,
  text: PropTypes.string.isRequired,
  image: PropTypes.string.isRequired,
  updated: PropTypes.object.isRequired,
};
