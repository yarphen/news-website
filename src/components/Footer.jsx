import React from 'react';
import { Link } from 'react-router-dom';

export const Footer = () => (
  <div className="footer-wrapper">
    <div className="footer">
      <div className="footer-left">
        <div className="footer-left-item">
          <div className="footer-left-item-wrapper">
            <Link to="/news">News</Link>
          </div>
        </div>
        <div className="footer-left-item">
          <div className="footer-left-item-wrapper">
            <Link to="/regions">Regions</Link>
          </div>
        </div>
        <div className="footer-left-item">
          <div className="footer-left-item-wrapper">
            <Link to="/video">Video</Link>
          </div>
        </div>
        <div className="footer-left-item">
          <div className="footer-left-item-wrapper">
            <Link to="/tv">TV</Link>
          </div>
        </div>
      </div>
      <div className="footer-right">Copyright &copy; yarphen</div>
    </div>
  </div>
);
