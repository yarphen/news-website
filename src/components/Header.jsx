/* eslint-disable jsx-a11y/click-events-have-key-events */

import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import { Search } from './Search';

class Header extends PureComponent {
  constructor(props) {
    super(props);
    this.handleOpenSidebar = this.handleOpenSidebar.bind(this);
  }

  handleOpenSidebar() {
    const { onOpenSidebar } = this.props;
    onOpenSidebar();
  }

  render() {
    return (
      <div className="header-wrapper">
        <Link to="/"><div className="big-header-logo fas fa-registered" /></Link>
        <div className="mobile-header">
          <div className="header-title">News</div>
          <Link to="/"><div className="header-logo fas fa-registered" /></Link>
          <div className="header-sandwich fas fa-bars" onClick={this.handleOpenSidebar} />
        </div>
        <div className="nav-menu">
          <div className="nav-menu-item"><Link to="/news">News</Link></div>
          <div className="nav-menu-item"><Link to="/regions">Regions</Link></div>
          <div className="nav-menu-item"><Link to="/video">Video</Link></div>
          <div className="nav-menu-item"><Link to="/tv">TV</Link></div>
          <Search />
        </div>
      </div>
    );
  }
}

Header.propTypes = {
  onOpenSidebar: PropTypes.func,
};

Header.defaultProps = {
  onOpenSidebar: () => {},
};

export { Header };
