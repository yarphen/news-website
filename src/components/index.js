export * from './ContentItem';
export * from './LoadMore';
export * from './Sidebar';
export * from './Header';
export * from './Footer';
export * from './ClearFilters';
export * from './Search';
