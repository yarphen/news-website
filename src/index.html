<!DOCTYPE html>
<html>

<head>
  <meta charset="UTF-8">
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  <meta http-equiv="X-UA-Compatible" content="ie=edge">
  <link rel="manifest" href="/manifest.json" />
  <script async src="https://cdn.jsdelivr.net/npm/pwacompat@2.0.6/pwacompat.min.js"
    integrity="sha384-GOaSLecPIMCJksN83HLuYf9FToOiQ2Df0+0ntv7ey8zjUHESXhthwvq9hXAZTifA"
    crossorigin="anonymous"></script>
</head>

<body>
<div id="root">

</div>
</body>

</html>