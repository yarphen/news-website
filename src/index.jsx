import React from 'react';
import ReactDOM from 'react-dom';
import { Route, Switch } from 'react-router-dom';
import { Provider } from 'react-redux';
import { ConnectedRouter } from 'connected-react-router';

import { store, history } from './store';
import { MyContainer, NotFound } from './containers';

import './css/app.css';

ReactDOM.render(
  <Provider store={store}>
    <ConnectedRouter history={history}>
      <Switch>
        <Route path="/" exact component={MyContainer} />
        <Route path="*" component={NotFound} />
      </Switch>
    </ConnectedRouter>
  </Provider>,
  document.getElementById('root'),
);
