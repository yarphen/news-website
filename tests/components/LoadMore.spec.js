import React from 'react';
import { create } from 'react-test-renderer';
import { LoadMore } from '../../src/components';

describe('LoadMore component', () => {
  test('it matches the snapshot', () => {
    const component = create(<LoadMore onLoadMore={() => {}} />);
    expect(component.toJSON()).toMatchSnapshot();
  });
});
