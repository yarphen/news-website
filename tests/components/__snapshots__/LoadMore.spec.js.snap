// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadMore component it matches the snapshot 1`] = `
<div
  className="load-more"
>
  <div
    className="load-more-btn"
    onClick={[Function]}
  >
    Load More
  </div>
</div>
`;
